### PR TITLE
feat: upgrade next to 14 and remove tf dependency

### PR DESCRIPTION
**Pull request by NithinKuruba**

### CODE REVIEW

File: app/controllers/keycloak.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
-import { RoleMappingPayload } from 'keycloak-admin/lib/defs/roleRepresentation';
+import ClientRepresentation from 'keycloak-admin/lib/defs/clientRepresentation';
```
Comment (jlangy) [inline app/controllers/keycloak.ts:1]: This lib is archived now, I wonder if it makes sense while we have time to try out the new one. Probably a different ticket though

**Review by jlangy — COMMENTED** [review]
Looks good, just left a couple of comments and questions

File: app/controllers/keycloak.ts [hunk 2]
```diff
@@ -127,3 +130,288 @@ export const addUserAsRealmAdmin = async (username: string, envs: string[], real
...
+    const realm = await kcAdminClient.realms.findOne({ realm: realmName });
+    if (realm === null) {
+      // create custom realm
+      const realm = await kcAdminClient.realms.create({
+        realm: realmName,
+        enabled: true,
+      });
+      if (realm) {
+        // fetch created custom realm
+        const customRealm = await kcAdminClient.realms.findOne({ realm: realmName });
```
[the remark this answers is not in the document]
Comment (NithinKuruba) [reply app/controllers/keycloak.ts:147]: Yeah the function just returns name of the realm so I have to fetch it again.

File: app/controllers/keycloak.ts [hunk 3]
```diff
@@ -127,3 +130,288 @@ export const addUserAsRealmAdmin = async (username: string, envs: string[], real
...
+  } catch (err) {
+    console.error(err);
+    throw Error(`Failed to create openid client: ${clientConfig.clientId}`);
+  }
+};
+
+const createGroup = async (realmName: string, env: string, groupName: string): Promise<GroupRepresentation> => {
+  const kcCore = new KeycloakCore(env);
+  const kcAdminClient = await kcCore.getAdminClient();
+  const groupExists = await kcAdminClient.groups.find({
```
[the remark this answers is not in the document]
Comment (NithinKuruba) [reply app/controllers/keycloak.ts:348]: Yeah I missed to update that var name, nice catch!

File: app/__tests__/api/delete-realm.test.ts [hunk 4]
```diff
@@ -89,67 +90,16 @@ describe('Realm Delete Request', () => {
...
-    const { req, res } = createMocks({
-      method: 'DELETE',
-      query: { id: 1 },
-    });
     mockAdminSession();
     await deleteHandler(req, res);
     expect(res.statusCode).toBe(200);
-
-    const rosterUpdateArgs = (prisma.roster.update as jest.Mock).mock.calls[0][0];
-    expect(rosterUpdateArgs.data.archived).toBe(true);
```
[the remark this answers is not in the document]
Comment (NithinKuruba) [reply app/__tests__/api/delete-realm.test.ts:118]: Somehow got deleted but yes restored them

**Review by jlangy — APPROVED** [review]

**Pull request merged**